tests/test_edgeql_calls.py

```diff
@@ -1243,6 +1243,36 @@ async def test_edgeql_calls_35(self):
             ]
         )
 
+    @test.xfail('''
+       Fails with 'ISE: cannot cast type record to <stuff>', I think
+       because the subcomponents need to be cast also.
```

Comment:
Yeah, this is an annoying Postgres quirk: it refuses to cast composite types even if they're binary-compatible!  We are handling this elsewhere, e.g:

https://github.com/edgedb/edgedb/blob/6891b88ad419709dad5fea7b4e9c7281ec2f9019/edb/edgeql/compiler/viewgen.py#L707-L727

But we forget to do the same for function returns here:

https://github.com/edgedb/edgedb/blob/6891b88ad419709dad5fea7b4e9c7281ec2f9019/edb/schema/functions.py#L1428-L1436

(`implicitly_castable` does not necessarily mean that it is implicitly castable in Postgres, which is what `is_type_compatible` checks)

Reply:
Going to merge and then will try to address in a follow-up